docs/mine/mitigating-dos.md

```diff
@@ -0,0 +1,109 @@
+iptables -A INPUT -p TCP -m multiport --dports 22,12345 -m state --state NEW -j ACCEPT
+```
+
+At the end of setting these rules, save them with:
+
+```
+# for Ubuntu
+sudo /sbin/iptables-save
+
+# for Red Hat / CentOS –
+sudo /sbin/service iptables save
+# or
+sudo  /etc/init.d/iptables save 
+```
+
+## IP based protection
+
+If you detect incoming traffic from an IP address, you can block all incoming traffic coming from that location. 
+
+```
+# Blocking 192.168.1.100 to exemplify.
+iptables -I INPUT -s 192.168.1.100 -j DROP
+```
+
+After setting this rule, remember to save your new rule with
+
+```
+# for Ubuntu
+sudo /sbin/iptables-save
+
+# for Red Hat / CentOS –
+sudo /sbin/service iptables save
+# or
+sudo  /etc/init.d/iptables save 
+```
+
+Learn more how to maintain a denylist of IPs at [Blocking IP Addresses in Linux with IPTables](https://linux-audit.com/blocking-ip-addresses-in-linux-with-iptables)
+
+## libp2p based protection
+
+If a malicious actor has successfully dialed to your miner and is spamming it at the Gossipsub level, you can block it at the libp2p level.
+
+To detect this, look for patterns such as:
+
+```
+2020-08-31T16:12:16.214-0400	INFO	pubsub	go-libp2p-pubsub@v0.3.5/pubsub.go:927	received message from router graylisted peer 12D3KooWFf6t5Zj4u8585r9bLwU3unykX4DmHftUtx6kEzw89anL. Dropping RPC
+2020-08-31T16:12:16.214-0400	INFO	pubsub	go-libp2p-pubsub@v0.3.5/pubsub.go:927	received message from router graylisted peer 12D3KooWJp8Kj9w65eQcsEdfT4dfhsWAeiroMfxAdZCzqd2wbHq2. Dropping RPC
+2020-08-31T16:12:16.215-0400	INFO	pubsub	go-libp2p-pubsub@v0.3.5/pubsub.go:927	received message from router graylisted peer 12D3KooWJp8Kj9w65eQcsEdfT4dfhsWAeiroMfxAdZCzqd2wbHq2. Dropping RPC
+```
+
+In this case, it is likely that a node is misbehaving and sending so many invalid messages that it was already discarded from the Gossipsub routing table. To disable it from successfully setting a libp2p connection, you can use the following command:
+
+```
+TODO: ADD COMMAND HERE
+```
+
+You can also block this peer IPs by fetching its multiaddrs and adding a rule for the IP addresses on the iptables. Use:
+
+```
+lotus net findpeer 12D3KooWFf6t5Zj4u8585r9bLwU3unykX4DmHftUtx6kEzw89anL
+TODO: ADD EXAMPLE HERE OF THE OUTPUT
+```
+
+And then follow the instructions on [IP based protection](#ip-based-protection) to block these IPs.
+
+## Separating the Miner from the Gatekeeping machine
+
+In order to be more effective when blocking incoming traffic, we recommend using a separate machine to front all the incoming traffic, so that the processing of IP packets, which can be an expensive operation when under attack, is not done by the Miner which also needs to run some CPU bound operations.
+
+Additionally, you can also consider setting up a Bastion to avoid direct access to your Miner machines. This will enable you to better control who has access and when operators access these machines.
+
+![](/images/filecoin-miner-operation.png)
```

Comment:
Best to use a relative link to this image too, then it will be hashed and included at build time making it IPFS compatible when using a gateway path to access the docs.

Reply:
p.s. It's ok for images to live alongside the markdown so everything is more portable. The `/docs/mine/img` folder might be the best place in this case.